pkg/ebpf/c/lock_contention.c

```diff
@@ -107,24 +108,20 @@ static __always_inline int record_pcpu_freelist_locks(u32 fd, struct bpf_map* bm
         if (!region)
             return -EINVAL;
 
-        struct lock_range lr_pcpu_lock = {
-            .addr_start =  region,
-            .range = sizeof(struct pcpu_freelist_head),
-            .type = HASH_PCPU_FREELIST_LOCK,
-        };
-
+        struct lock_range lr_pcpu_lock = {};
+        lr_pcpu_lock.addr_start = region;
+        lr_pcpu_lock.range = sizeof(struct pcpu_freelist_head);
+        lr_pcpu_lock.type = HASH_PCPU_FREELIST_LOCK;
         err = bpf_map_update_elem(&map_addr_fd, &lr_pcpu_lock, &mapid, BPF_NOEXIST);
         if (err < 0)
             return err;
     }
 
     // this regions contains the lock htab->freelist.extralist.lock
-    struct lock_range lr_global_lock = {
-        .addr_start = (u64)(&htab->freelist),
-        .range = sizeof(struct pcpu_freelist),
-        .type = HASH_GLOBAL_FREELIST_LOCK,
-    };
-
+    struct lock_range lr_global_lock = {};
+    lr_global_lock.addr_start = (u64)(&htab->freelist);
+    lr_global_lock.range = sizeof(struct pcpu_freelist);
+    lr_global_lock.type = HASH_GLOBAL_FREELIST_LOCK;
```

Comment:
Curious, why is this not equivalent to the old code? Do we need to do this for other structs as well?

Reply:
The old code did not initialize the padding and the verifier would complain.